waku/v2/node/waku_node.nim

```diff
@@ -507,43 +507,40 @@ proc unsubscribe*(node: WakuNode, pubsubTopic: PubsubTopic, contentTopics: Conte
 ## Waku archive
 
 proc mountArchive*(node: WakuNode,
-                   driver: Option[ArchiveDriver],
-                   messageValidator: Option[MessageValidator],
-                   retentionPolicy: Option[RetentionPolicy]) =
+                   storeMessageDbUrl: string,
+                   storeMessageDbVacuum: bool = false,
+                   storeMessageDbMigration: bool = false,
+                   storeMessageRetentionPolicy: string = "none"):
```

Comment:
The node does not receive a "DB" it receives an Archive as provided by the application - it should not have to have any knowledge about this external component. This may in fact not even be a DB, but simply a chunk of memory.